### PR TITLE
Skip or hide section on Plugin Page

### LINKED ISSUE #2277: Skip the required/suggested plugins page when all requirements are met.
**Is your feature request related to a problem? Please describe.**
When creating a report, the plugin page is always shown, even if all required and optional plugins are already enabled. This can be confusing for the user, because it implies that a plugin isn't enabled yet.

See issue #2007

**Describe the solution you'd like**
- Skip the plugin page if all required and optional plugins are enabled
- Hide the required plugin section if all required plugins are enabled (but not all optional plugins)

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
This PR makes some changes to the plugin page (Setup page) while creating a report. 

The code already included the following scenario:
* If all **optional** plugins are enabled, this section is hidden on the plugin page.

This PR adds the following scenario's:
* If all **required** plugins are enabled, this section is hidden on the plugin page.
* If **both** all required and optional plugins are enabled, the entire plugin page is skipped (and the report is shown).

These changes have been implemented in the following tree reports:
- Generate report
- Aggregate report
- Multi-report

### Issue link
Closes #2277 

### Demo
Only shows optional:
afbeelding

Skips entire page - video:
https://github.com/minvws/nl-kat-coordination/assets/99282220/3c2715e6-4959-4770-b2ab-d7547f1a2181

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
The list of selected plugins on the resulting page seems to be empty. 
We should probably make this list show the boefjes that are both (suggested or suggested) and enabled;

set(required boefjes + suggested boefjes)  ∩ set(enabled boefjes)

**Comment by madelondohmen** [comment]
> The list of selected plugins on the resulting page seems to be empty. We should probably make this list show the boefjes that are both (suggested or suggested) and enabled;
> 
> set(required boefjes + suggested boefjes) ∩ set(enabled boefjes)

Should be fixed now.

**Comment by underdarknl** [comment]
Code looks good to me.

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Looks good, haven't identified bugs. Can confirm that the sections are either shown or hidden depending on the enabled boefjes. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

File: rocky/reports/views/base.py [hunk 1]
```diff
@@ -109,11 +111,16 @@ def get_report_types_for_aggregate_report(
...
 
-    def get_required_optional_plugins(self, plugin_ids: Dict[str, Set[str]]) -> Dict[str, Plugin]:
+    def get_required_optional_plugins(self, plugin_ids: Dict[str, Set[str]]) -> (Dict[str, Plugin], Dict[str, bool]):
         plugins = {}
         for plugin, plugin_ids in plugin_ids.items():
             plugins[plugin] = [get_katalogus(self.organization.code).get_plugin(plugin_id) for plugin_id in plugin_ids]
-        return plugins
+
+        all_plugins_enabled = {
+            plugin_type: all(plugin.enabled for plugin in plugin_list) for plugin_type, plugin_list in plugins.items()
```
Comment (ammar92) [inline rocky/reports/views/base.py:120]: I think you can also put the entire generator in `all`, potentially enhancing its laziness by causing it to stop at the first falsy value

**Review by ammar92 — APPROVED** [review]
Just a simple suggestion, no other remarks

File: rocky/reports/views/base.py (hunk 1 above)
Comment (ammar92) [reply rocky/reports/views/base.py:120]: The original expression checks if all plugins across all plugin lists are enabled, while the suggestion expression checks if all plugins within each individual plugin list are enabled. So the current implementation is already correct 👍

**Pull request merged**